### PR TITLE
[breaking change] new api with areEqual

### PR DESCRIPTION
**Pull request by dai-shi**
1. make types strict (no-any)
2. a suggestion to change the api: `args: any[]` -> `arg: any`
3. custom areEqual with Object.is by default. deepEqual optional.

Hope you like it!

- [x] impl
- [x] readme
- [ ] examples

### CODE REVIEW

**Comment by drcmda** [comment]
how would it look like in code examples, could you illustrate the differences? we must put them into the readme.md docs anyway @dai-shi

**Comment by dai-shi** [comment]
Do you mean examples on codesandbox? I wish we have codesandbox ci configured for this repo so that forking them are easy.
I will work on readme in the mean time.

**Comment by dai-shi** [comment]
@drcmda readme updated!

File: package.json [hunk 1]
```diff
@@ -52,6 +50,7 @@
     "@types/react-test-renderer": "^16.9.3",
     "@typescript-eslint/eslint-plugin": "^3.7.1",
     "@typescript-eslint/parser": "^3.7.1",
+    "fast-deep-equal": "^3.1.3",
```
Comment (rijk) [inline package.json:53]: Should this still be here? It seems like the import is removed in this change.

Comment (dai-shi) [reply package.json:53]: No, we don't need it here. I was assuming we might have examples in this repo, but all examples are on codesandbox, so we can remove it.

**Comment by drcmda** [comment]
thanks again! used this as a reference for https://github.com/pmndrs/use-asset/commit/d1cff93a22722e3ecc6dd89e0d0db1365f6ff546 in order to keep ...args

**Pull request closed without merging**